### PR TITLE
Fix/send metrics

### PR DESCRIPTION
**Pull request by rholshausen**
* Fixes the metrics call by making the send metrics function async and getting the verification wait on it
* Adds a ConsumerTestRun event to the pactffi_cleanup_mock_server function.

Not sure if the pactffi_cleanup_mock_server is the best place for that event.

### CODE REVIEW

File: rust/pact_ffi/src/mock_server/mod.rs [hunk 1]
```diff
@@ -419,7 +421,27 @@ pub extern fn pactffi_mock_server_mismatches(mock_server_port: i32) -> *mut c_ch
...
+      let interactions = match mock_server {
+        Either::Left(ms) => {
+          let pact = ms.pact.lock().unwrap();
+          pact.interactions().len()
+        },
+        Either::Right(ms) => ms.pact.interactions.len()
+      };
+      send_metrics(MetricEvent::ConsumerTestRun {
+        interactions,
+        test_framework: "pact_ffi".to_string(),
```
Comment (mefellows) [inline rust/pact_ffi/src/mock_server/mod.rs:434]: Should these be coming from the client library, so we can disambiguate languages? I think we do this on the provider side right?

**Review by mefellows — APPROVED** [review]
Just had the clarification about passing through the client framework so we can track languages separately. Other than that, it looks OK to my eyes

File: rust/pact_ffi/src/mock_server/mod.rs (hunk 1 above)
Comment (rholshausen) [reply rust/pact_ffi/src/mock_server/mod.rs:434]: Yes, but there is no way to get that info from within the `pactffi_cleanup_mock_server` function.

Comment (mefellows) [reply rust/pact_ffi/src/mock_server/mod.rs:434]: Right, yes that makes sense. We can always add later

**Pull request merged**